validator/CourseValidator.py

```diff
@@ -0,0 +1,117 @@
+from persistence.postgre import DB
+from exceptions.CourseException import *
+
+COURSE_SUBSCRIPTION = {"basico": 1, "estandar": 2, "premium": 3}
+USER_SUBSCRIPTION = {"free": 1, "platinum": 2, "black": 3}
+
+
+class CourseValidator:
+    def __init__(self, database: DB):
+        self.db = database
+
+    def hasACourseWithTheSameName(self, courseName, creatorId):
+        for course in self.db.getCourses({"creator_id": creatorId}):
+            if course.get("name", "") == courseName:
+                return True
+        return False
+
+    def courseExists(self, courseId: int):
+        return self.db.getCourse(courseId) is not None
+
+    def isTheCourseCreator(self, courseId: int, userId: int):
+        return userId == self.db.getCourse(courseId)["creator_id"]
+
+    def isACollaborator(self, courseId: int, collaboratorId: int):
+        filter = {"subscribers": False}
+        return collaboratorId in self.db.getUsers(courseId, filter)
+
+    def isSubscribed(self, courseId: int, subscriberId: int):
+        filter = {"subscribers": True}
+        return subscriberId in self.db.getUsers(courseId, filter)
+
+    def canSubscribe(self, courseId: int, userData: dict):
+        courseData = self.db.getCourse(courseId)
+        if (
```

Comment:
return not (...)
Es innecesario mandar dos ifs para devolver lo que la condicion booleana ya te dice